kafka-connect-hazelcast/src/main/scala/com/datamountaineer/streamreactor/connect/hazelcast/config/HazelCastSinkSettings.scala

```diff
@@ -65,25 +66,38 @@ object HazelCastSinkSettings {
       }
     }).toMap
 
-    val fieldsMap = routes.map(
-      rm => (rm.getSource, rm.getFieldAlias.map(fa => (fa.getField,fa.getAlias)).toMap)
+    val fieldMap = routes.map(
+      rm => (rm.getSource, rm.getFieldAlias.map( fa => (fa.getField,fa.getAlias)).toMap)
     ).toMap
 
     val ignoreFields = routes.map(r => (r.getSource, r.getIgnoredField.toSet)).toMap
     val groupName = config.getString(HazelCastSinkConfig.SINK_GROUP_NAME)
     require(groupName.nonEmpty,  s"No ${HazelCastSinkConfig.SINK_GROUP_NAME} provided!")
     val connConfig = HazelCastConnectionConfig(config)
+    val client = HazelCastConnection.buildClient(connConfig)
     val format = routes.map(r => (r.getSource, getFormatType(r.getFormatType))).toMap
-
-    new HazelCastSinkSettings(groupName,
-                              connConfig,
-                              routes,
-                              topicTables,
-                              fieldsMap,
-                              ignoreFields,
-                              errorPolicy,
-                              maxRetries,
-                              format)
+    val p = routes.map(r => (r.getSource, r.getPrimaryKeys.toSet)).toMap
+
+
+    //get the field expected in the sink record which maps to a primary key
+    val pks = fieldMap.map({
```

Comment:
why p above is not enough? you are only picking the fields for which there is an alias and that alias has been set alias?